src/benchmark/benchmark_activeobjectmgr.cpp

```diff
@@ -0,0 +1,111 @@
+// Minetest
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include "benchmark_setup.h"
+#include "server/activeobjectmgr.h"
+#include "util/numeric.h"
+
+namespace {
+
+class TestObject : public ServerActiveObject {
+public:
+	TestObject(v3f pos) : ServerActiveObject(nullptr, pos)
+	{}
+
+	ActiveObjectType getType() const {
+		return ACTIVEOBJECT_TYPE_TEST;
+	}
+	bool getCollisionBox(aabb3f *toset) const {
+		return false;
+	}
+	bool getSelectionBox(aabb3f *toset) const {
+		return false;
+	}
+	bool collideWithObjects() const {
+		return true;
+	}
+};
+
+constexpr float POS_RANGE = 2001;
+
+inline v3f randpos()
+{
+	return v3f(myrand_range(-POS_RANGE, POS_RANGE),
+		myrand_range(-20, 60),
+		myrand_range(-POS_RANGE, POS_RANGE));
+}
+
+inline void fill(server::ActiveObjectMgr &mgr, size_t n)
+{
+	mgr.clear();
+	for (size_t i = 0; i < n; i++) {
+		auto obj = std::make_unique<TestObject>(randpos());
+		bool ok = mgr.registerObject(std::move(obj));
+		REQUIRE(ok);
+	}
+}
+
+}
+
+template <size_t N>
+void benchGetObjectsInsideRadius(Catch::Benchmark::Chronometer &meter)
+{
+	server::ActiveObjectMgr mgr;
+	size_t x;
+	std::vector<ServerActiveObject*> result;
+
+	auto cb = [&x] (ServerActiveObject *obj) -> bool {
+		x += obj->m_static_exists ? 0 : 1;
+		return false;
+	};
+	fill(mgr, N);
+	meter.measure([&] {
+		x = 0;
+		mgr.getObjectsInsideRadius(randpos(), 30.0f, result, cb);
```

Comment:
isn't a very small radius also a common case, for finding a static entity belonging to a node?

e.g. signs_lib: https://github.com/mt-mods/signs_lib/blob/3d5ba32fa0db60a2741778ed91743411a9bdaa5c/api.lua#L206

Reply:
Is is, but I don't think that it will produce categorically different results (even on a possible future data structure).